### PR TITLE
Fix interfaces inheritance support

### LINKED ISSUE #3034: Interfaces inheritance
I'm doing some generic work on several unrelated entities with common code, thanks to interfaces.

linq2db works well enough with interfaces but it seems to ignore members that come from interfaces hierarchies.

For example if you declare this model:
```cs
interface IA
{
  int Id { get; set; }
}

interface IB : IA
{
 string Name { get; set; }
}
```

If you then write Linq that refers to `Id` through `IB`, such as:
```cs
public void ChangeName(int id, string name)
{
  return db.GetTable<IB>().TableName("MyTable")
    .Where(x => x.Id == id)
    .Set(x => x.Name, x => newName)
    .Update();
}
```

It compiles but at runtime linq2db complains that `x.Id` is not a mapped property of `IB`.

### PR DESCRIPTION
**Pull request by MaceWindu**
Fix  #3034

Cherrypick from #4036

### CODE REVIEW

File: Source/LinqToDB/Extensions/ReflectionExtensions.cs [hunk 1]
```diff
@@ -62,6 +65,35 @@ public static MemberInfo[] GetPublicInstanceValueMembers(this Type type)
...
 
+		private static MemberInfo[] GetInterfacePublicInstanceValueMembers(Type type)
+		{
+			var members = type
+				.GetMembers(BindingFlags.Instance | BindingFlags.Public)
+				.Where(m => m.IsFieldEx() || m.IsPropertyEx() && ((PropertyInfo)m).GetIndexParameters().Length == 0);
+
+			var interfaces = type.GetInterfaces();
+			if (interfaces.Length == 0)
+				return members.ToArray();
```
Comment (viceroypenguin) [inline Source/LinqToDB/Extensions/ReflectionExtensions.cs:76]: Should this be possible since `type` is already known to be an Interface?

**Review by viceroypenguin — DISMISSED** [review]

File: Source/LinqToDB/Extensions/ReflectionExtensions.cs (hunk 1 above)
Comment (MaceWindu) [reply Source/LinqToDB/Extensions/ReflectionExtensions.cs:76]: Yes, for interface it returns inherited interfaces

**Review by igor-tkachev — APPROVED** [review]

**Pull request merged**